src/interface/obsidian/src/chat_modal.ts

```diff
@@ -337,17 +360,12 @@ export class KhojChatModal extends Modal {
                     referenceExpandButton.innerHTML = expandButtonText;
                     references.appendChild(referenceSection);
                 } else {
-                    if (responseText.startsWith("{") && responseText.endsWith("}")) {
-                    } else {
-                        // If the chunk is not a JSON object, just display it as is
-                        continue;
-                    }
-
-                    this.renderIncrementalMessage(responseElement, responseText);
+                    await this.renderIncrementalMessage(responseElement, responseText);
                 }
             }
         } catch (err) {
-            this.renderIncrementalMessage(responseElement, "Sorry, unable to get response from Khoj backend ❤️‍🩹. Contact developer for help at team@khoj.dev or <a href='https://discord.gg/BDgyabRM6e'>in Discord</a>")
+            let errorMsg = "<p>Sorry, unable to get response from Khoj backend ❤️‍🩹. Contact developer for help at team@khoj.dev or <a href='https://discord.gg/BDgyabRM6e'>in Discord</a></p>";
```

Comment:
Rather than adding a selective rendering for "raw html", you could also just set this default message to be in markdown format.
```suggestion
            let errorMsg = "Sorry, unable to get response from Khoj backend ❤️‍🩹. Contact developer for help at [team@khoj.dev](mailto:team@khoj.dev) or [in Discord](https://discord.gg/BDgyabRM6e)";
```

Reply:
Done